docs/source/Objective_Functions/Objective_Function.md

```diff
@@ -1,25 +1,248 @@
+
+```
+objective_func_channel = await Channel.create(client=client)
+local_opt_channel = await Channel.create(client=client)
+
+tetra_opt_parameters = {
+    "dimensionality": 2,
+    "iteration_number": 2,
+    "maximal_rank": 4,
+    "points_number": 1,
+    "quantization": False,
+    "tolerance": 0.0010000000474974513,
+    "lower_limits": [0, 0],
+    "upper_limits": [9, 9],
+    "grid": [10, 10],
+     "objective_function_channel_id": objective_func_channel
+     #local_optimizer_channel_id parameter is optional
+    "local_optimizer_channel_id": local_opt_channel
+    "polling": {"initial_delay": 1.0, "retries": 100, "delay": 1.0, "backoff_factor": 1.1}
+}
+```
+
+An example of TetraOpt parameters using `an https endpoint`. Notice the
+`objective_function` and `local_optimizer parameter`:
+```
+tetra_opt_parameters = {
+    "dimensionality": 2,
+    "iteration_number": 2,
+    "maximal_rank": 4,
+    "points_number": 1,
+    "quantization": False,
+    "tolerance": 0.0010000000474974513,
+    "lower_limits": [0, 0],
+    "upper_limits": [9, 9],
+    "grid": [10, 10],
+    'objective_function':'http://127.0.0.1:8000/test_func_eval/Ackley/',
+     #local_optimizer parameter is optional  
+    "local_optimizer": "http://127.0.0.1:8000/local_optimization/Ackley/",
+    "polling": {"initial_delay": 1.0, "retries": 100, "delay": 1.0, "backoff_factor": 1.1}
+}
+```
+
+### 1. To use the communications channel:
+The channel uses objects of the following classes to send and receive information:
+1. `Ask`
+2. `Tell`
+
+It also need the following class to stream the information to TetraOpt
+1. `Channel`
+
+We can import them from the tq42 library in our Python script using the following:
+1. `from tq42.channel import Channel, Ask, Tell`
+
+
+#### The `Ask` Class
+In our example above, we used the Ackley function provided by the OptimizationTestFunctions library as our objective function. 
+
+An Example of an `Ask` object values passed by TetraOpt:
+```
+{
+    parameters {
+      values: 1
+      values: 3
+    }
+    parameters {
+      values: 2
+      values: 7
+    }
+    headers: "h0"
+    headers: "h1"
+}
+```
+
+The fields are formally described in Google Protocol Buffers as:
+```
+message Ask {
+  repeated Parameter parameters = 1 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).repeated.min_items = 1
+  ];
+  repeated string headers = 2;
+}
+
+message Parameter {
+  repeated float values = 1 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).repeated.min_items = 1
+  ];
+}
+
+```
+#### The `Tell` Class 
+To return the answers to TetraOpt, we need to create a Tell object. For the objective function, we need the list of the results, the parameters and the headers to construct it. For the local optimization function, we need an extra candidates parameter.
+
+An Example of `Tell` object values for an objective and local optimization function:
+```
+{
+    parameters {
+      values: 1
```

Comment:
again please use json for readability for these examples. This does not help an avid engineer / data scientist.